### PR TITLE
Implement container attach

### PR DESCRIPTION
**Pull request by jwhonce**
Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM, but we really need @baude to review this.

**Comment by giuseppe** [comment]
_containers_attach.py LGTM

**Comment by umohnani8** [comment]
LGTM

File: contrib/python/test/test_containers.py [hunk 1]
```diff
@@ -65,8 +63,22 @@ def test_get(self):
...
     def test_attach(self):
-        with self.assertRaisesNotImplemented():
-            self.alpine_ctnr.attach()
+        # StringIO does not support fileno() so we had to go old school
+        input = os.path.join(self.tmpdir, 'test_attach.stdin')
+        output = os.path.join(self.tmpdir, 'test_attach.stdout')
+
+        with open(input, 'w+') as mock_in, open(output, 'w+') as mock_out:
+            # double quote is indeed in the expected place
+            mock_in.write('echo H"ello, World"; exit\n')
```
Comment (wking) [inline contrib/python/test/test_containers.py]: Minor nit, but you probably meant `"H` instead of `H"`.

**Comment by rhatdan** [comment]
@jwhonce Fix @wking and then we can merge.

File: contrib/python/test/podman_testcase.py [hunk 2]
```diff
@@ -62,7 +62,8 @@ def run_cmd(*args):
 
         cmd = ['podman']
         cmd.extend(podman_args)
-        cmd.extend(['run', '-d', 'alpine', 'sleep', '500'])
+        # cmd.extend(['run', '-d', 'alpine', 'sleep', '500'])
```
Comment (baude) [inline contrib/python/test/podman_testcase.py]: did you mean to keep this ?

File: contrib/python/test/test_containers.py (hunk 1 above)
Comment (baude) [reply contrib/python/test/test_containers.py]: I think based on his comment, he has it where he wants it?

**Comment by baude** [comment]
optional nit, LGTM

File: contrib/python/test/test_containers.py (hunk 1 above)
Comment (wking) [reply contrib/python/test/test_containers.py]: > I think based on his comment, he has it where he wants it?

Ah, I somehow missed the comment.  Out of curiosity, why put it there?

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

File: contrib/python/test/test_containers.py (hunk 1 above)
Comment (jwhonce) [reply contrib/python/test/test_containers.py]: @wking I wanted the target in the return stream to be the words 'Hello, World' from the echo command not the characters being printed from the shell 'H"ello, World'

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
Forcing merge. Since HOMU seems a little hosed up.

**Pull request merged**